File: javascript/two_pointers/344. Reverse String.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
    let l = 0, r = s.length - 1;

    while (l < r) {
        temp = s[r];
        s[r] = s[l];
        s[l] = temp;

        l++;
        r--;
    };
};

File: python/two_pointers/344. Reverse String.py
'''
1. two pointer (own)
Time: O(n)
Space: O(1)
2. s.reverse()
Time: O(n)
Space: O(1)
3. stack
Time: O(n)
Space: O(n)
'''

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        l, r = 0, len(s) - 1

        while l < r:
            s[l], s[r] = s[r], s[l]

            l += 1
            r -= 1